sessions/react-state-3/journal-app-filter/README.md

```diff
@@ -0,0 +1,103 @@
+# React State: Journal App with Entries Filter
+
+Your journal app includes filter button to show all entries or only favorites. Currently the button don't work yet. Let's implement this feature!
+
+> 💡 You can use this template as a starting point. But if you are far enough along with your own Journal App, please use that instead.
+
+## Task
+
+### Make the filter buttons clickable (optional)
+
+> 💡 If you didn't write a wrapping component like `Tab` for the `button` tag in your own code, you might not need to do this extra step.
+
+Users should be able to click on the filter buttons to change the list of displayed entries. First we want to make sure the buttons are clickable.
+
+Locate the component that renders the `button` JSX tag. In the template there is `Tab` used as a wrapping component around the `button` tag. To have the button clickable we need to pass the `onClick` prop to it.
+
+- Add the prop `onClick` to the function declaration of this component
+- Pass the prop to the `button` tag like this: `onClick={onClick}`
```

Comment:
Writing an additional `handel*` function for this would be overengineering. We need to make the `<button>` JSX tag, which is nested in `Tab`, clickable by adding a `onClick` to it. We can just pass it down as it is from the `Tab` component. Another `handle*` function would just be harder to read and two lines of code that don't provide any benefit.

It's meant like this:

```jsx
function Tab({children, onClick}){
  return <button onClick={onClick}>{children}</button>
}

function Tabs() {
  return (
    <>
      <Tab onClick={handleWhatever}>Yeah!</Tab>
      <Tab onClick={handleThisThing}>Awesome!</Tab>
    </>
  )
}
```

Maybe we should rephrase the task?

Reply:
I don't think re-writing is necessary at all, was just curious how this would be handled (hee hee).

Reply:
Okay :)